### PR TITLE
Add narrow no-break space support

### PR DESCRIPTION
**Pull request by quentin-bettoum**
The goal of this PR is to support the rendering of narrow no-break spaces as asked in #6640.

Note that I'm not proficient in Rust, but since it looked simple, I gave it a try by looking at what have been done for regular `nbsp`.

The tests passes and the feature works using `cargo run`.

I don't know what's better between having a dedicated `editor.whitespace.render` option for `nnbsp` or including it in the existing `nbsp` parameter.

Currently, the new config parameters would look like this:
```toml
[editor.whitespace.render]
space = "all"
nbsp = "all"
nnbsp = "all"
tab = "all"
newline = "none"

[editor.whitespace.characters]
space = "·"
nbsp = "⍽"
nnbsp = "␣"
tab = "→"
newline = "⏎"
tabpad = "·"

```

### CODE REVIEW

File: helix-core/src/chars.rs
Comment (the-mikedavis) [inline helix-core/src/chars.rs:1]: Looks like this change is unrelated - could you reset this?

**Review by the-mikedavis — APPROVED** [review]

**Review by archseer — APPROVED** [review]
